Assets/Scripts/Robo.cs

```diff
@@ -131,7 +132,13 @@ private IEnumerator Iframes()
         curr_health = Mathf.Clamp(curr_health - damage, 0, max_health);
         curr_health = health_bar.UpdateHealth(curr_health);
 
-        StartCoroutine(Iframes());
+        if(this.curr_health == 0){
```

Comment:
Actually just for safety let's make this a leq check:
```suggestion
        if(this.curr_health <= 0){
```

Reply:
`curr_health = health_bar.UpdateHealth(curr_health);` clamps it below by 0, so in the current implementation it's definitely not a risk that curr_health will be negative.

You trying to future-proof it against our possible future mistakes? If you think it's worth, I'm down to commit the suggestion.

Reply:
> You trying to future-proof it against our possible future mistakes? If you think it's worth, I'm down to commit the suggestion.

Initially yes but the more I think about it the more I feel like the brittle check of exact equality might be more helpful, since if something does go wrong with clamping (or any step between clamping and the check) the more general check may obfuscate that error. I'll resolve this with no change for now.